### DISCUSSION TITLE
Generate routes unexpected indentifier

### DISCUSSION THREAD

**Discussion by jeffreyvdhondel** [nuxt/nuxt#17442] [Questions]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-27506609.eed7036`
- Package Manager:  `yarn@1.22.18`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

```
import { defineNuxtConfig } from "nuxt3";
import { $fetch } from "ohmyfetch";

const dynamicRoutes = async (): Promise<string[]> => {
  return $fetch<
    {
      userId: number;
      id: number;
      title: string;
      complted: boolean;
    }[]
  >("https://jsonplaceholder.typicode.com/todos").then((data) => {
    return data.map((x) => `/${x.title.replace(/\s/g, "-")}`);
  });
};

// https://v3.nuxtjs.org/docs/directory-structure/nuxt.config
export default defineNuxtConfig({
  typescript: {
    strict: true,
  },
  target: "static",
  generate: {
    routes: await dynamicRoutes(),
  },
});
```

### Describe the bug

After reading the documentatie of the generate routes function i tried to implement this see code example but when ran i get the following error : 

```
 ERROR  Unexpected identifier                                                                                                                                                   08:01:35

  routes: await dynamicRoutes() } });exports.default = _default;
  ^^^^^^^^^^^^^
  
  SyntaxError: Unexpected identifier
  at new Script (node:vm:100:7)
  at createScript (node:vm:257:10)
  at Object.runInThisContext (node:vm:305:10)
  at jiti (node_modules/jiti/dist/jiti.js:1:196321)
  at resolveConfig (node_modules/c12/dist/index.mjs:179:18)
  at loadConfig (node_modules/c12/dist/index.mjs:97:40)
  at async loadNuxtConfig (node_modules/@nuxt/kit/dist/index.mjs:625:59)
  at async loadNuxt (node_modules/nuxt3/dist/index.mjs:1362:19)
  at async loadNuxt (node_modules/@nuxt/kit/dist/index.mjs:657:19)
  at async Object.invoke (node_modules/nuxi/dist/chunks/build.mjs:28:18)
```

**Comment by danielroe** [comment]
That's a syntax error, because `await` is only usable in an async function. It is recommended to use a Nuxt hook or module if you need an async function.

**Reply by jeffreyvdhondel** [reply]
Ok but was the hook for adding dynamic routes based on an api response?

**Reply by jeffreyvdhondel** [reply]
Tried it with the following but the console log wont trigger: 

```
export default defineNuxtConfig({
  typescript: {
    strict: true,
  },
  target: "static",
  hooks: {
    "generate:route": ({ route }) => {
      console.log(route);
    },
  },
});
```

**Comment by matthewbecker1990** [comment]
@jeffreyvdhondel  Did you manage to solve this issue? I'm trying to get the same thing working currently.

**Reply by jeffreyvdhondel** [reply]
No, Still nothing i think currently you cant add dynamic/static routes based on a API response i'm looking into the "pages:extend" route to generate the pages that are available on the api.

**Reply by danielroe** [reply]
You could do this via the `nitro:config` hook or simply add them to `nuxt.options.generate.routes` within a module...
